src/test/transaction_tests.cpp

```diff
@@ -157,24 +221,28 @@ BOOST_AUTO_TEST_CASE(tx_valid)
             BOOST_CHECK(state.IsValid());
 
             PrecomputedTransactionData txdata(tx);
-            for (unsigned int i = 0; i < tx.vin.size(); i++)
-            {
-                if (!mapprevOutScriptPubKeys.count(tx.vin[i].prevout))
-                {
-                    BOOST_ERROR("Bad test: " << strTest);
-                    break;
-                }
+            std::vector<unsigned int> verify_flags = ParseScriptFlags(test[2].get_str());
 
-                CAmount amount = 0;
-                if (mapprevOutValues.count(tx.vin[i].prevout)) {
-                    amount = mapprevOutValues[tx.vin[i].prevout];
-                }
-                unsigned int verify_flags = ParseScriptFlags(test[2].get_str());
-                const CScriptWitness *witness = &tx.vin[i].scriptWitness;
-                BOOST_CHECK_MESSAGE(VerifyScript(tx.vin[i].scriptSig, mapprevOutScriptPubKeys[tx.vin[i].prevout],
-                                                 witness, verify_flags, TransactionSignatureChecker(&tx, i, amount, txdata), &err),
-                                    strTest);
-                BOOST_CHECK_MESSAGE(err == SCRIPT_ERR_OK, ScriptErrorString(err));
+            // Do not use TrimFlags() in the main test in order to detect invalid verifyFlags combination in the test file.
+            if (!CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, ~verify_flags[0], txdata, strTest, /* expect_valid */ true))
+                BOOST_ERROR("Tx unexpectedly failed: " << strTest);
+            // Check that flags are maximal: transaction should fail if any unset flags are set.
+            for (size_t i = 0; i < mapFlagNames.size(); i++)
+            {
+                // Backwards compatibility: Removing a flag should not invalidate a valid transaction
+                unsigned int flags = TrimFlags(~(verify_flags[0] | (1U << i)));
+                if (!CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, flags, txdata, strTest, /* expect_valid */ true))
+                    BOOST_ERROR("Tx unexpectedly failed with flag " << ToString(i) << " unset: " << strTest);
+                // Backwards compatibility: Removing some random flags should not invalidate a valid transaction
+                flags = TrimFlags(~(verify_flags[0] | (unsigned int)InsecureRandBits(mapFlagNames.size())));
+                if (!CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, flags, txdata, strTest, /* expect_valid */ true))
+                    BOOST_ERROR("Tx unexpectedly failed with random flags " << ToString(flags) << ": " << strTest);
+            }
+            for (size_t f = 1; f < verify_flags.size(); f++)
+            {
```

Comment:
join

Reply:
@glozow if you run clang-format on your changes, it can avoid the need for all the formatting feedback

Reply:
oops 😅 thanks for advice @jonatack

Reply:
https://github.com/bitcoin/bitcoin/blob/master/contrib/devtools/README.md#clang-format-diffpy